unit_test/solver/network/assets/test_fall_type.py

```diff
@@ -0,0 +1,290 @@
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Test FallType entities."""
+import unittest
+from unittest.mock import patch
+from uuid import uuid4
+
+import numpy as np
+import numpy.testing as np_testing
+
+from simulator_core.solver.matrix.core_enum import NUMBER_CORE_QUANTITIES, IndexEnum
+from simulator_core.solver.network.assets.fall_type import FallType
+from simulator_core.solver.network.assets.node import Node
+
+
+class FallTypeTest(unittest.TestCase):
+    """Testcase for Boundary class."""
+
+    def setUp(self) -> None:
+        """Set up the test case."""
+        # Create a BaseBoundary object
+        self.asset = FallType(
+            name=uuid4(),
+        )
+        # Create supply, connection_point:0 and return node, connection_point:1
+        self.supply_node = Node(name=uuid4())
+        self.return_node = Node(name=uuid4())
+        # Connect the nodes to the asset
+        self.asset.connect_node(node=self.supply_node, connection_point=0)
+        self.asset.connect_node(node=self.return_node, connection_point=1)
+
+    @patch.object(FallType, "add_thermal_equations")
+    @patch.object(FallType, "add_press_to_node_equation")
+    @patch.object(FallType, "add_internal_cont_equation")
+    @patch.object(FallType, "add_internal_pressure_loss_equation")
+    def test_get_equations(
+        self, press_loss_eq_patch, internal_cont_eq_patch, press_to_node_patch, thermal_patch
+    ) -> None:
+        """Evaluate the retrieval of equations from the boundary object."""
+        # Arrange
+
+        # Act
+        equations = self.asset.get_equations()  # act
+
+        # Assert
+        self.assertEqual(thermal_patch.call_count, 2)
+        self.assertEqual(press_to_node_patch.call_count, 2)
+        self.assertEqual(internal_cont_eq_patch.call_count, 1)
+        self.assertEqual(press_loss_eq_patch.call_count, 1)
+        self.assertEqual(len(equations), 6)
+
+    def test_get_equation_insufficient_nodes(self) -> None:
+        """Evaluate the retrieval of equations from the boundary object with insufficient nodes."""
+        # Arrange
+        self.asset.connected_nodes = []
+
+        # Act
+        with self.assertRaises(ValueError) as cm:
+            self.asset.get_equations()
+
+        # Assert
+        self.assertIsInstance(cm.exception, ValueError)
+        self.assertEqual(str(cm.exception), "The number of connected nodes must be 2!")
+
+    def test_get_equations_more_unknowns(self) -> None:
+        """Evaluate the retrieval of equations from the boundary object with more unknowns."""
+        # Arrange
+        self.asset.number_of_unknowns = 4
```

Comment:
Dit si ook minder dan 6

Reply:
Renamed test to invalid number of unknowns

Reply:
heb je deze al gepushed?

Reply:
Ja alleen heb ik de test behouden tot wij de fix doorvoeren voor _number_of_unknowns als private.